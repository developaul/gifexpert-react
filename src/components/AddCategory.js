import React, { useState } from 'react';
import PropTypes from 'prop-types';

import shortid from 'shortid';

const AddCategory = ({ categories, setCategory }) => {
    
    const [ inputValue, setInputValue ] = useState( '' );

    const handleInputChange = e => setInputValue( e.target.value );

    const handleSubmit = e => {
        e.preventDefault();
    
        const [{ nameCategory }] = categories;

        if( inputValue.trim().length > 2 && inputValue.trim() !== nameCategory ) {

            setCategory( cats => [{
                nameCategory: inputValue.trim(),
                id: shortid.generate()
            }, ...cats ] );

            setInputValue( '' );
        }
    };

    return (
        <form
            onSubmit={ handleSubmit }
        >
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    );
};

AddCategory.propTypes = {
    categories: PropTypes.array.isRequired,
    setCategory: PropTypes.func.isRequired
};

export default AddCategory;