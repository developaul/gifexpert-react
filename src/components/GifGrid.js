import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem, Spinner } from './';

const GifGrid = ({ nameCategory }) => {

    const { data:images, loading } = useFetchGifs( nameCategory );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{ nameCategory }</h3>

            { loading && <Spinner /> }

            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id }
                            nameCategory={ nameCategory }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    nameCategory: PropTypes.string.isRequired
};

export default GifGrid;