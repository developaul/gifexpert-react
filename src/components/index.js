import AddCategory from './AddCategory';
import GifGrid from './GifGrid';
import GifGridItem from './GifGridItem';

export {
    AddCategory,
    GifGrid,
    GifGridItem
}