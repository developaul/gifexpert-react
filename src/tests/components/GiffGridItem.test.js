import React from 'react';

import { shallow } from 'enzyme';

import { GifGridItem } from '../../components';

describe( 'Pruebas en GifGridItem', () => {
    
    const nameCategory  = 'Name Category Gif',
          title         = 'Un titulo',
          url           = 'htpps://localhost/algo.jpg';
        
    const wrapper = shallow( 
        <GifGridItem  
            nameCategory={ nameCategory } 
            title={ title }
            url={ url }
        />
    );

    test( 'Debe de mostrar correctamente el componente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'Debe de tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find( 'p' );
        
        expect( p.text() ).toBe( title );

    });

    test( 'Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find( 'img' );

        const { src, alt } =  img.props();

        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    });

    test( 'Debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find( 'div' );

        expect( div.hasClass( 'animate__fadeIn' ) ).toBe( true );

    });
    
});