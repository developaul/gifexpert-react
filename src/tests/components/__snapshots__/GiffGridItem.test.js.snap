// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="htpps://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
